transport/http/outbound.go

```diff
@@ -434,10 +434,10 @@ func checkServiceMatch(reqSvcName string, resHeaders http.Header) (bool, string)
 //
 // OpenTracing information must be added manually, before this call, to support context propagation.
 func (o *Outbound) RoundTrip(hreq *http.Request) (*http.Response, error) {
-	return o.roundTrip(hreq, nil /* treq */, time.Now())
+	return o.roundTrip(hreq, nil /* treq */, time.Now(), true /* transparent proxying*/)
 }
 
-func (o *Outbound) roundTrip(hreq *http.Request, treq *transport.Request, start time.Time) (*http.Response, error) {
+func (o *Outbound) roundTrip(hreq *http.Request, treq *transport.Request, start time.Time, transparentProxy bool) (*http.Response, error) {
```

Comment:
if `bool` is not very clear, I'm open to use a state enum. But I think we only have 2 states here.

Reply:
Instead of a `bool` the caller can just pass the `sender`. You can store the `&transportSender{...}` that wraps a client on the `*Outbound`, that way the call site will be either passing in `o.client` or `o.transport`